opencog/reasoning/RuleEngine/rules/pln/evaluation-to-member.scm

```diff
@@ -0,0 +1,64 @@
+; ============================================================================= 
+; GeneralEvaluationToMemberRule
+;	EvaluationLink (pred D (ListLink B C))
+;		becomes
+;	MemberLink( B (SatisfyingSetLink( Variable $X 
+;		(EvaluationLink (pred D (ListLink X C))))))
+; -----------------------------------------------------------------------------
+
+(define pln-rule-evaluation-to-member
+	(BindLink
+		(ListLink
+			(VariableNode "$B")
+			(VariableNode "$C")
+			(VariableNode "$D")
+			(TypedVariableLink
+				(VariableNode "$D")
+				(VariableTypeNode "PredicateNode")))
+		(ImplicationLink
+			(EvaluationLink
+				(VariableNode "$D")
```

Comment:
It sounds like this solution won't be using the RuleEngine at all?  The new RuleEngine PLN design uses BindLink.

I imagined another hacky approach:

  1. Wrap the undefined number of arguments as one untyped VariableNode like

     ```
  EvaluationLink
    (VariableNode "$D")  ; PredicateNode type
    (VariableNode "$P")  ; no type restriction
     ```

  2. Passes (VariableNode "$P") to the helper function `scm:pln-formula-evaluation-to-member` and construct the new MemberLink in the helper instead.

In my mind it seems that would work, but I haven't tried it.  Anyone got a better idea?